src/features-service.ts

```diff
@@ -52,14 +62,22 @@ export class FeaturesService {
 
     features.available_firmware_version = _.find(FIRMWARE_METADATA_FILE, {modelNumber: features.model}).version;
 
+    let bootloaderHash: string = features.bootloader_mode ? '' : features.bootloader_hash.toHex();
+    let isUnofficialBootloader = !features.bootloader_mode && !_.includes(OFFICIAL_BOOTLOADER_HASHES, bootloaderHash);
```

Comment:
Is it safe to assume that the bootloader is always valid while in bootloader mode? isUnofficialBootloader is always false while in bootloader mode

Reply:
The bootloader hash isn't available in bootloader mode, so there isn't a good way to know.